src/wasm-type.h

```diff
@@ -174,10 +174,12 @@ class Type {
   // Returns the feature set required to use this type.
   FeatureSet getFeatures() const;
 
-  // Gets the heap type corresponding to this type
+  // Gets the heap type corresponding to this type, assuming that it is a
+  // reference or Rtt type.
   HeapType getHeapType() const;
 
-  const struct Rtt& getRtt() const;
+  // Gets the Rtt for this type, assuming that it is an Rtt type.
+  Rtt getRtt() const;
```

Comment:
What's the benefit to removing `const&`? At least the const part seems nice to clarify that this does not return an item to be modified.

Reply:
There's no concrete benefit, but since #3428 we have a documented convention of passing `Rtt`s by value. Once you're returning by value, I don't think the `const` has any effect.

Reply:
Isn't there a small benefit to disallowing `getRtt().depth = 5` at compile time?

Reply:
I just checked locally and that is disallowed with or without the `const` with "error: expression is not assignable."

Reply:
Oh nice, thanks for checking!